src/main/java/net/neoforged/neoforge/client/settings/KeyMappingLookup.java

```diff
@@ -31,13 +32,60 @@ public class KeyMappingLookup {
      * @return the list of key mappings
      */
     public List<KeyMapping> getAll(InputConstants.Key keyCode) {
+        return this.getAll(keyCode, false);
+    }
+
+    /**
+     * Returns all active keys associated with the given key code and the active
+     * modifiers and conflict context.
+     *
+     * @param keyCode   the key being pressed
+     * @param releasing if the key is being released
+     * @return the list of key mappings
+     */
+    public List<KeyMapping> getAll(InputConstants.Key keyCode, boolean releasing) {
         List<KeyMapping> matchingBindings = new ArrayList<>();
-        KeyModifier activeModifier = KeyModifier.getActiveModifier();
-        // Apply active modifier only if the pressed key is not the modifier itself
-        // Otherwise, look for key bindings without modifiers
-        if (activeModifier == KeyModifier.NONE || activeModifier.matches(keyCode) || !matchingBindings.addAll(findKeybinds(keyCode, activeModifier))) {
+        // Get a list of all active modifiers
+        List<KeyModifier> activeModifiers = KeyModifier.getActiveModifiers();
+
+        for (var modifier : activeModifiers) {
+            // If modifier matches, add other modifiers
+            if (modifier.matches(keyCode)) {
+                // Check if binding matches with another modifier
+                for (var key : activeModifiers) {
+
+                    // Skip if modifier matches
+                    if (key.matches(keyCode)) {
+                        continue;
+                    }
```

Comment:
Couldn't this be replaced with a `key == modifier` condition instead? That more strongly gives the first-glance impression of "loop on all modifiers except the originally matched one". Unless we anticipate that multiple `KeyModifier`s might match on the same key code in the future (as they are all exclusive to each other for now).

I ask this since this threw me for a loop (ha) trying to understand what this nested loop was doing. Only after a few rereads did I realize what it does (at least, what it seems to me).